OpenKeychain/src/main/java/org/sufficientlysecure/keychain/securitytoken/OpenPgpCapabilities.java

```diff
@@ -0,0 +1,154 @@
+/*
+ * Copyright (C) 2016 Nikita Mikhailov <nikita.s.mikhailov@gmail.com>
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package org.sufficientlysecure.keychain.securitytoken;
+
+import org.sufficientlysecure.keychain.util.Iso7816TLV;
+
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Map;
+
+public class OpenPgpCapabilities {
+    private final static int MASK_SM = 1 << 7;
+    private final static int MASK_KEY_IMPORT = 1 << 5;
+    private final static int MASK_ATTRIBUTES_CHANGABLE = 1 << 2;
+
+    private boolean mPw1ValidForMultipleSignatures;
+    private byte[] mAid;
+    private byte[] mHistoricalBytes;
+
+    private boolean mHasSM;
+    private boolean mAttriburesChangable;
+    private boolean mHasKeyImport;
+
+    private byte mSMAlgo;
+    private int mMaxCmdLen;
+    private int mMaxRspLen;
+
+    private Map<KeyType, KeyFormat> mKeyFormats;
+
+    public OpenPgpCapabilities(byte[] data) throws IOException {
+        Iso7816TLV[] tlvs = Iso7816TLV.readList(data, true);
```

Comment:
There is a lot of logic in this constructor, and it throws an IOException which is not very nice. All fields are determined entirely in the constructor if I'm not missing something, so moving the parsing logic into a static method and making all fields final would yield a nice immutable class here.